File: leecode/code/739/739.java
class Solution {

    public int[] dailyTemperatures(int[] T) {
        int[] record = new int[T.length];

        for (int i = 0; i < T.length; i++) {
            int temp = 0;
            for (int j = i + 1; j < T.length; j++) {
                if (T[j] > T[i]) {
                    temp = j - i;
                    break;
                }
            }
            record[i] = temp;
        }
        return record;
    }
}

File: leecode/code/739/739.py
class Solution:
    def dailyTemperatures(self, T: List[int]) -> List[int]:
        rec = list()
        res = [0] * len(T)


        for i in range(len(T) - 1, -1, -1):
            while len(rec) > 0 and T[rec[len(rec) - 1]] <= T[i]:
                rec.pop()
            
            if len(rec) == 0:
                res[i] = 0
            else:
                res[i] = rec[len(rec) - 1] - i
            
            rec.append(i)
            
        
        return res